packer/linux/scripts/install-docker.sh

```diff
@@ -6,28 +6,46 @@ DOCKER_RELEASE="stable"
 DOCKER_COMPOSE_VERSION=1.28.6
 MACHINE=$(uname -m)
 
-# This performs a manual install of Docker.
-
 # Add docker group
 sudo groupadd docker
 sudo usermod -a -G docker ec2-user
 
-# Manual install ala https://docs.docker.com/engine/installation/binaries/
-curl -Lsf -o docker.tgz "https://download.docker.com/linux/static/${DOCKER_RELEASE}/${MACHINE}/docker-${DOCKER_VERSION}.tgz"
-tar -xvzf docker.tgz
-sudo mv docker/* /usr/bin
-rm docker.tgz
+# Add the Docker yum repository, but override the url to get CentOS 7 not
+# CentOS 2 (the default with Amazon Linux 2 applied to the interpolation)
+sudo yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+sudo yum-config-manager --setopt="docker-ce-stable.baseurl=https://download.docker.com/linux/centos/7/$MACHINE/stable" --save
+
+# Add CentOS 7 Base and Extras to the yum repos list, needed for deps for
+# packages from the Docker yum repo
+cat << 'EOF' | sudo tee /etc/yum.repos.d/centos.repo
+[base]
+name=CentOS-7 - Base
+mirrorlist=http://mirrorlist.centos.org/?release=7&arch=$basearch&repo=os
+enabled=1
+gpgcheck=1
+priority=1
+
+[extras]
+name=CentOS-7 - Extras
+mirrorlist=http://mirrorlist.centos.org/?release=7&arch=$basearch&repo=extras
+enabled=1
+gpgcheck=1
+priority=1
+EOF
+
+import_rpm_key() {
+  local key="${1}"
+  local url="${2}"
+  sudo curl --fail --location --output "/etc/pki/rpm-gpg/${key}" "${url}"
+  sudo rpm -import "/etc/pki/rpm-gpg/${key}"
+}
+
+import_rpm_key "RPM-GPG-KEY-CentOS-7" "https://centos.org/keys/RPM-GPG-KEY-CentOS-7"
+import_rpm_key "RPM-GPG-KEY-CentOS-7-aarch64" "https://centos.org/keys/RPM-GPG-KEY-CentOS-7-aarch64"
 
-sudo mkdir -p /etc/docker
-sudo cp /tmp/conf/docker/daemon.json /etc/docker/daemon.json
-sudo cp /tmp/conf/docker/subuid /etc/subuid
-sudo cp /tmp/conf/docker/subgid /etc/subgid
-sudo chown -R ec2-user:docker /etc/docker
+# Do the install
+sudo yum install -y -q "docker-ce-${DOCKER_VERSION}" "docker-ce-cli-${DOCKER_VERSION}" containerd.io
```

Comment:
This is lifted from the Docker install docs with the version suffixes added, probably don’t need the literal call out to `containerd.io` which is likely a dep. Can test again without it.